test_conformance/subgroups/test_workgroup_non_uniform_vote.cpp

```diff
@@ -474,31 +335,173 @@ int test_work_group_functions_non_uniform_vote(cl_device_id device,
     run_for_type rft(device, context, queue, num_elements, true,
                      required_extensions);
 
-    int error = rft.run_nu_all_eq<cl_char>();
```

Comment:
There is a lot of code duplication below. If the mask is no longer a template argument, a for loop could help here. Otherwise, I would recommend having function(s) to remove the code duplication. For example,

```c++
template <unsigned mask> int run_nu_all_eq(run_for_type rft) {
  int error = rft.run_nu_all_eq<cl_char, mask>();
  error |= rft.run_nu_all_eq<cl_uchar, mask>();
  ...
  return error;
}
```

There are other alternatives to shorten this function, I suppose.

Reply:
WorkGroupParams struct solve this issue